pytorch-optuna.ipynb

```diff
@@ -342,39 +342,6 @@
     "### Start Dask cluster with GPU workers"
    ]
   },
-  {
-   "cell_type": "markdown",
-   "id": "aad974d9-f81c-465a-9090-7fa40b9ef91d",
-   "metadata": {},
-   "source": [
-    "First we need to define a software environment.\n",
-    "\n",
-    "Usually Dask does this for us automatically by scraping our local environment, but in this case our local laptop doesn't have a GPU or GPU-specific libraries, so we'll need to create a remote software environment explicitly."
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "30a5d06b-a292-447a-b333-0e6c6e99b035",
-   "metadata": {
-    "tags": []
-   },
-   "outputs": [],
-   "source": [
-    "%%time\n",
-    "\n",
-    "import coiled\n",
-    "\n",
-    "coiled.create_software_environment(\n",
-    "    name=\"pytorch\",\n",
-    "    conda={\n",
-    "        \"channels\": [\"pytorch\", \"nvidia\", \"conda-forge\", \"defaults\"],\n",
-    "        \"dependencies\": [\"python=3.11\", \"dask=2023.5.1\", \"pytorch\", \"optuna\", \"torchvision\", \"cudatoolkit\", \"pynvml\"],\n",
-    "    },\n",
-    "    gpu_enabled=True,\n",
-    ")"
-   ]
-  },
```

Comment:
I'm inclined to keep this actually. I think that it's more useful in the context of a Dask example. There's already a bunch of stuff they have to do, so adding one more step doesn't seem as bad, and the education seems good here.

Same below probably.  Thoughts?

Reply:
I'm confused now. If we would want to keep this, why did we have James make a docker image?

Are you saying you want `coiled.create_software_environment(name=..., container=...)`?

Reply:
I don't have a strong view but just noting that generally people (e.g. KoBold) really like the `container=` argument in `coiled.Cluster`, rather than having to make a software environment.

Reply:
I want a docker image for situations like Coiled notebooks docs where creating a software environment is a lot heavier-weight than the command we want them to run. 

I prefer Coiled software environments over docker images for most users (you and I saw this on Monday for example with Ameya I think).  However, sometimes I want to be able to skip that conversation, for example in a getting started guide for GPU notebooks.  In those cases I'd really like to have a docker image handy.  

Does that make sense?  Other confusions?

Reply:
I agree with what you're saying about docker images vs manual software environments being situation dependent. When I've presented these notebooks, I've definitely glossed over the software environment part (unless someone asks about it, then I go into more details). I think this approach has been well received. @mrocklin have you been taking a different approach? 

While there is a lot of content in the `pytorch*` notebooks, most of that content is `pytorch`-related code. I like being able to say "This is a bunch of `pytorch` code that will look familiar to `pytorch` users. To have this code run on GPUs on our cluster, we just need to ask for instance types with GPUs".

Reply:
When I present the notebooks I definitely gloss over the software environment part (as well as 90% of the notebook).  It's precisely because I'm able to gloss over that detail that I'm ok with it taking up a bunch of space.  

However, if someone goes to the next step, of using these examples not as a presentation device, but as an example to copy-paste-modify from, then we want them to have an easy way to add packages.  

I think that if we switch to docker then we make the presentation mildly smoother (but only mildly, because we're glossing over so much already) but substantially diminish the value of the examples as something that users can leverage themselves.

Reply:
OK.  I'll add the second constraint then that we do have some example people can play with to help the use PyTorch with Coiled.  If this isn't that example that's fine, but can we make another one then?

My hope is that we can integrate these examples more into the docs so that they become those ready-to-use examples.